components/ai_chat/core/browser/model_service.cc

```diff
@@ -212,6 +214,13 @@ base::Value::Dict GetModelDict(mojom::ModelPtr model) {
       kCustomModelItemApiKey,
       EncryptAPIKey(model->options->get_custom_model_options()->api_key));
   model_dict.Set(kCustomModelItemKey, model->key);
+
+  std::optional<uint32_t> context_size =
+      model->options->get_custom_model_options()->context_size;
+  if (context_size.has_value()) {
+    model_dict.Set(kCustomModelContextSizeKey, static_cast<int>(*context_size));
```

Comment:
this will potentially truncate the context length, I think

Reply:
The context length is not likely to exceed the low millions anytime soon. This should be low enough for ints and uint32s AFAIK. That said, if there's a better approach, I'm all ears. Eager to learn :)

Reply:
Going to wrap this in a validity check to make sure the value is clamped. Wanted to store a uint32_t in the Dict, but found that is not possible. Casting to an int/int32_t, with bounds-checking, should be okay I think.